lib/chewy/index/actions.rb

```diff
@@ -150,45 +150,63 @@ def #{method} options = {}
           METHOD
         end
 
-        # Deletes, creates and imports data to the index.
-        # Returns import result
+        # Deletes, creates and imports data to the index. Returns the
+        # import result. If index name suffix is passed as the first
+        # argument - performs zero-downtime index resetting.
         #
-        #   UsersIndex.reset!
-        #
-        # If index name suffix passed as the first argument - performs
-        # zero-downtime index resetting (described here:
-        # http://www.elasticsearch.org/blog/changing-mapping-with-zero-downtime/).
+        # It also applies journal if anything was journalled during the
+        # reset.
         #
+        # @example
+        #   UsersIndex.reset!
         #   UsersIndex.reset! Time.now.to_i
         #
-        def reset!(suffix = nil, journal: false, **options)
-          result = if suffix.present? && (indexes = self.indexes).present?
+        # @see http://www.elasticsearch.org/blog/changing-mapping-with-zero-downtime
+        # @param suffix [String] a suffix for the newly created index
+        # @param apply_journal [true, false] journal is applied after the import is completes
+        # @param journal [true, false] journallig is switched of for import during reset by default
+        # @param import_options [Hash] options, passed to the import call
+        # @return [true, false] false in case of errors
```

Comment:
It doesn't. And I found this confusing as well, don't remember, why did I decide to use bang here, probably because it is a really destructive action. But I've decided not to touch it in this PR and do something later maybe